pkg/spanconfig/spanconfigreconciler/reconciler.go

```diff
@@ -435,6 +448,43 @@ func (f *fullReconciler) deleteExtraneousSpanConfigs(
 	return extraneousTargets, nil
 }
 
+// dropInvalidPTS records inspects the PTS system table and drops any invalid
+// PTS records.
+func dropInvalidPTSRecords(
+	ctx context.Context, ptsStore protectedts.Storage, sourceTenant roachpb.TenantID,
+) error {
+	ptsState, err := ptsStore.GetState(ctx)
```

Comment:
this loads all protected timestamps into memory right? this feels like a heavy handed read call to do during every incremental reconciliation. To zoom out, do we need to dropInvalidPTSRecords during incremental reconciliation? If the only way to end up with these invalid records is post tenant restore or post PCR cutover, will we always run a full reconciliation loop first, which will catch these invalid records?

Reply:
Hmm, although I guess if the incremental reconciler were to fail instead, [the job would retry](https://github.com/kev-cao/cockroach/blob/27829e0dec9c167f69c8cbfac38832effb007375/pkg/spanconfig/spanconfigjob/job.go#L139-L141) `Reconcile`.

So it would run the full reconciliation process again.